app/scripts/controllers/transactions/index.js

```diff
@@ -1877,6 +1881,49 @@ export default class TransactionController extends EventEmitter {
     this.memStore.updateState({ unapprovedTxs, currentNetworkTxList });
   }
 
+  _calculateTransactionsCost(txMeta, approvalTxMeta) {
+    let approvalGasCost = '0x0';
+    if (approvalTxMeta?.txReceipt) {
+      approvalGasCost = calcGasTotal(
+        approvalTxMeta.txReceipt.gasUsed,
+        approvalTxMeta.txReceipt.effectiveGasPrice,
+      );
+    }
+    const tradeGasCost = calcGasTotal(
+      txMeta.txReceipt.gasUsed,
+      txMeta.txReceipt.effectiveGasPrice,
+    );
+    const tradeAndApprovalGasCost = new BigNumber(tradeGasCost, 16)
+      .plus(approvalGasCost, 16)
+      .toString(16);
+    return {
+      approvalGasCostInEth: Number(
+        conversionUtil(approvalGasCost, {
+          fromNumericBase: 'hex',
+          toNumericBase: 'dec',
+          fromDenomination: 'WEI',
+          toDenomination: 'ETH',
+        }),
+      ),
+      tradeGasCostInEth: Number(
+        conversionUtil(tradeGasCost, {
+          fromNumericBase: 'hex',
+          toNumericBase: 'dec',
+          fromDenomination: 'WEI',
+          toDenomination: 'ETH',
+        }),
+      ),
+      tradeAndApprovalGasCostInEth: Number(
+        conversionUtil(tradeAndApprovalGasCost, {
+          fromNumericBase: 'hex',
+          toNumericBase: 'dec',
+          fromDenomination: 'WEI',
+          toDenomination: 'ETH',
+        }),
```

Comment:
Not blocking, but we could create a helper function since all of the params to `conversionUtil` are the same.  Something like:

```
function calculateGasCost(cost) {
  return Number(conversionUtil(cost, {
          fromNumericBase: 'hex',
          toNumericBase: 'dec',
          fromDenomination: 'WEI',
          toDenomination: 'ETH',
  })
}
```

Reply:
We actually already have a helper function for this conversion: [hexWeiToDecETH](https://github.com/MetaMask/metamask-extension/blob/07da8ce58923cc3fc0959691c9f04c5b1312891e/ui/helpers/utils/conversions.util.js#L156)

Reply:
Nice!